receiver/vcenterreceiver/scraper_test.go

```diff
@@ -32,14 +33,37 @@ import (
 
 func TestScrape(t *testing.T) {
 	ctx := context.Background()
-	mockServer := mock.MockServer(t)
+	mockServer := mock.MockServer(t, false)
 
 	cfg := &Config{
 		Metrics:  metadata.DefaultMetricsSettings(),
 		Endpoint: mockServer.URL,
 		Username: mock.MockUsername,
 		Password: mock.MockPassword,
 	}
+
+	testScrape(t, ctx, cfg)
+}
+
+func TestScrape_Tls(t *testing.T) {
+	ctx := context.Background()
+	mockServer := mock.MockServer(t, true)
+
+	cfg := &Config{
+		Metrics:  metadata.DefaultMetricsSettings(),
+		Endpoint: mockServer.URL,
+		Username: mock.MockUsername,
+		Password: mock.MockPassword,
+		TLSClientSetting: configtls.TLSClientSetting{
+			Insecure:           true,
+			InsecureSkipVerify: true,
+		},
+	}
+
+	testScrape(t, ctx, cfg)
+}
+
+func testScrape(t *testing.T, ctx context.Context, cfg *Config) {
```

Comment:
```suggestion
func testScrape(ctx context.Context, t *testing.T, cfg *Config) {
```

Linter yells at you for
> scraper_test.go:66:31: context-as-argument: context.Context should be the first parameter of a function (revive)
func testScrape(t *testing.T, ctx context.Context, cfg *Config) {